quickwit/quickwit-index-management/src/garbage_collection.rs

```diff
@@ -219,9 +229,32 @@ async fn delete_splits(
     while let Some(delete_split_result) = delete_split_from_index_res_stream.next().await {
         match delete_split_result {
```

Comment:
if this makes sense ( I didn't dig into that), it might be useful to stop using `Result<Vec<WhateverTheseEntriesAre>, DeleteSplitError>` and just have a `DeleteSplitsOutcome` or `DeleteSplitsResultSummary` object that looks like `DeleteSplitError`.

Reply:
it would make sense if it wasn't for delete_index using `delete_splits_from_storage_and_metastore`. I think returning a struct, and checking if it contains 0 errors is worse than generally having a Result, for which we may merge branches in some code paths

Reply:
ok